report-flow-ustc:1.0.0, touying-flow:1.0.0 and touying-pres-ustc:0.1.0

Quaternijkon opened the pull request:
<!--
Thanks for submitting a package! Please read and follow the submission guidelines detailed in the repository's README and check the boxes below. Please name your PR as `name:version` of the submitted package.

If you want to make a PR for something other than a package submission, just delete all this and make a plain PR.
-->

I am submitting
- [x] a new package
- [ ] an update for a package

<!--
Please add a brief description of your package below and explain why you think it is useful to others. If this is an update, please briefly say what changed.
-->

Description: Explain what the package does and why it's useful.

Hey! I thought I had removed all unnecessary packages from Touying that I had imported redundantly.

The **touying-flow** is a template file created based on the Touying project, characterized by its simplicity and practicality. It integrates various features, making it convenient for users to learn and use Typst to create presentations. It is particularly suitable for academic exchanges in settings such as universities.

The **touying-ustc** is a template created based on the Touying project, specifically designed for the University of Science and Technology of China (USTC). It uses high-quality, officially recommended images, colors, and styles from the university, catering to the needs of the USTC user community.

The **lab-report-template-ustc** is a template designed for USTC students to complete assignments and submit lab reports (of course, it can easily be adapted for any school or organization). The template incorporates extensive visual design and content work tailored to its specific use case.

<!--
These things need to be checked for a new submission to be merged. If you're just submitting an update, you can delete the following section.
-->

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

<!--
The following box only needs to be checked for **template** submissions. If you're submitting a package that isn't a template, you can delete the following section. See the guidelines section about licenses in the README for more details.
-->
- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

OrangeX4 commented:
Thank you for contributing! I have a few comments that you can consider, but it's not mandatory. First of all, about naming, I think the naming of the template should be more creative. The name touying-flow doesn't sound so much like a template, but rather like a toolkit for touying's subordinates. Maybe you could consider something more creative, like touying-fluent? And touying-ustc is a bit too official, you can consider adding a creative prefix, like simpl in touying-simpl-hkustgz. And the theme name should be the same as the package name, e.g. touying-flow you should use a function like #show: flow-theme but not dewdrop-theme (of course, touying-fluent corresponds to fluent-theme). Also, would you be happy to upgrade touying-ustc to touying 0.5.3? It may take a bit of work, but it will give the user a better experience (which is not mandatory, of course).

Quaternijkon commented:
> Thank you for contributing! I have a few comments that you can consider, but it's not mandatory. First of all, about naming, I think the naming of the template should be more creative. The name touying-flow doesn't sound so much like a template, but rather like a toolkit for touying's subordinates. Maybe you could consider something more creative, like touying-fluent? And touying-ustc is a bit too official, you can consider adding a creative prefix, like simpl in touying-simpl-hkustgz. And the theme name should be the same as the package name, e.g. touying-flow you should use a function like #show: flow-theme but not dewdrop-theme (of course, touying-fluent corresponds to fluent-theme). Also, would you be happy to upgrade touying-ustc to touying 0.5.3? It may take a bit of work, but it will give the user a better experience (which is not mandatory, of course).

@OrangeX4 I named the template "flow" because during a sharing session, a senior mentioned that presentations should emphasize an overall "flow," . At the same time, "flow"（心流）  refers to a focused mental state, which aligns with the goal of maintaining concentration. Moreover, I felt that the name "touying-xxx" would not cause misunderstanding,   whereas all the current touying-xxx  (maybe?) are templates. I also clarified its purpose in the description.

As for the name "touying-ustc," I noticed there is a "touying-buaa" on universe, so I thought "touying-ustc" is a viable name. Regarding updating touying to version 0.5.3, I think the features in version 0.4.2 are already sufficient for the target user group, and it fully meets their needs for using this template. I have introduced Typst and Touying to people around me and recommended the templates I created. For routine academic presentations, touying:0.4.2 offers functionality on par with Beamer. Therefore, updating to 0.5.3 doesn't seem like an urgent or necessary task (though I will consider updating the interface to 0.5.3 in the future so I set the version of touying-ustc to 0.1.0).

Lastly, if you don't mind, I have already renamed "dewdrop" to "flow."

OrangeX4 commented:
Then I think touying-flow is ok, but for touying-ustc, you should still choose a more creative name. touying-buaa is an earlier package, and more specific rules have not been made at that time, you can refer to #908. Anyway, thanks to creating these two templates, they're beautiful.

Quaternijkon commented:
> Then I think touying-flow is ok, but for touying-ustc, you should still choose a more creative name. touying-buaa is an earlier package, and more specific rules have not been made at that time, you can refer to #908. Anyway, thanks to creating these two templates, they're beautiful.

Thanks for your advice, I've changed the name to 'touying-pres-ustc' but left the variant names unchanged (since they would become too long and meaningless). Is this acceptable, or should I update all the 'ustc-xxx' variants to 'pres-ustc-xxx' to align with the theme name?

OrangeX4 commented:
I think it's doable, and it's up to you, if you feel that brevity is more important. Of course, if you align all the function names, I think it's nice to have better consistency for the user.

elegaanz commented on packages/preview/lab-report-template-ustc/1.0.0/content/example/main.typ:
```diff
@@ -0,0 +1,477 @@
+#import "../../theme.typ": *
```
This import should be absolute (i.e `#import "@preview/...": *`).

Quaternijkon replied:
sorry, I forgot to change it.

elegaanz commented on packages/preview/lab-report-template-ustc/1.0.0/content/template/main.typ:
```diff
@@ -0,0 +1,50 @@
+#import "../../theme.typ": *
```
I'm not sure what the role of this file is?

Quaternijkon replied:
I think somebody may not want to edit example.typ directly, so I provide an empty template for them.

elegaanz replied:
The thing is, this file will be quite hard to discover for users of your package: only the `content/example` folder will be copied to their project, so unless they dig in the repository or in the local package cache (which most people don't do I would say), they will never see it.

In the end it is up to you to decide whether you want to keep these files or not, but I given how hard they would be to find I'm not sure they add much. Also, it is relatively easy to select all or part of the example file and delete it, if someone wants a more minimal template.

Quaternijkon replied:
Since I often modify the templates I use to meet my specific style requirements, I believe there will still be people checking the repository. 
In the final version I submitted, I removed unnecessary files, even though they were quite small.

elegaanz commented on packages/preview/touying-flow/1.0.0/content/template/title.typ:
```diff
@@ -0,0 +1,44 @@
+#import "@preview/touying-flow:1.0.0":*
```
Same for this file.

Quaternijkon replied:
If you think it's unnecessary, I will delete them.

elegaanz reviewed (commented):
For the lab-report-template-ustc package that you added, you would need to change the name to something more unique (so that someone else from your university can submit an alternative package without being disadvantaged). If this template is officially endorsed by your university staff, it is possible to keep the name as is, but it would require some verification first.

On a related note, I've seen that you package the logos of the university. Do you have the rights to distribute them?

elegaanz commented:
Thanks. :)
